src/code-locations.cxx

```diff
@@ -156,7 +156,7 @@ bool CodeLocations::process_source_location (char *location_txt, location_t * lo
 		location->frames.source[f].valid =
 		    strcmp (location->frames.source[f].file, UNRESOLVED) != 0 &&
 		    strcmp (location->frames.source[f].file, NOT_FOUND) != 0 &&
-		    location->frames.source[f].line >= 0;;
+		    location->frames.source[f].line > 0;
```

Comment:
I have some doubts about this comparison. Sometimes, Extrae cannot obtain the exact source code line number and it emits a 0. So >=0 may be a safer. Probably we could drop the last comparison because this is an unsigned, so it will be always >= 0.

Reply:
I guess the question is whether it is an error/invalid when Extrae does not obtain the exact source code line. But the initial issue arose with the `>=0` as it was flagged as tautological by g++.

Reply:
Do you mind removing line 159 then, please?